src/rpc/fees.cpp

```diff
@@ -96,6 +99,59 @@ static RPCHelpMan estimatesmartfee()
     };
 }
 
+static RPCHelpMan estimatefeewithforecasters()
+{
+    return RPCHelpMan{
+        "estimatefeewithforecasters",
+        "\nEstimates the approximate fee per kilobyte needed for a transaction to begin\n"
+        "confirmation within conf_target blocks if possible Uses virtual transaction size as defined\n"
+        "in BIP 141 (witness data is discounted).\n",
+        {
+            {"conf_target", RPCArg::Type::NUM, RPCArg::Optional::NO, "Confirmation target in blocks"},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "", {
+                {RPCResult::Type::NUM, "Low priority fee rate estimate (" + CURRENCY_UNIT + "/kvB)", /*optional=*/true, "fee rate estimate in " + CURRENCY_UNIT + "/kvB for economical users"},
+                {RPCResult::Type::NUM, "High priority fee rate estimate (" + CURRENCY_UNIT + "/kvB)", /*optional=*/true, "fee rate estimate in " + CURRENCY_UNIT + "/kvB for conservative users"},
+                {RPCResult::Type::STR, "Forecaster", /*optional=*/true, "the forecaster that provide the fee rate estimate"},
+                {RPCResult::Type::ARR, "errors", /*optional=*/true, "Errors encountered during processing (if there are any)", {
+                    {RPCResult::Type::STR, "", "error"},
+                }},
+            }},
+        RPCExamples{HelpExampleCli("estimatefeewithforecasters", "2") + HelpExampleRpc("estimatefeewithforecasters", "2")},
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            FeeEstimator& fee_estimator = EnsureAnyFeeForecasters(request.context);
+            const unsigned int targetBlocks = request.params[0].getInt<unsigned int>();
+            UniValue result(UniValue::VOBJ);
+            UniValue errors(UniValue::VARR);
+            // The estimator currently support only Fee estimators that provide estimate for transaction to confirm
+            // as soon as possible in one or two blocks.
+            // If the confirmation target is more than 2 fail early.
+            // This should change whenever we have a forecaster that can support higher target.
+            if (targetBlocks > 2) {
+                errors.push_back("estimatefeewithforecasters currently provide estimates for confirmation target 1 and 2 only");
+                result.pushKV("errors", errors);
+                return result;
+            }
```

Comment:
In [rpc]: create an rpc estimatefeewithforecasters :

I wonder whether ideally this error should be surfaced from the `fee_estimator` itself, and not here...?